cobbler/cobbler_collections/collection.py

```diff
@@ -126,9 +133,16 @@ def find(self, name: str = "", return_list: bool = False, no_errors=False,
             except:
                 return self.listing.get(kargs["name"], None)
 
+        if self.api.settings().lazy_start:
+            # Forced deserialization of the entire collection to prevent deadlock in the search loop
+            for obj_name in self.get_names():
+                obj = self.get(obj_name)
+                if obj is not None and not obj.inmemory:
+                    obj.deserialize()
+
```

Comment:
This code led to an endless loop when using `mac_address`.
I think we should also backport #3654.

Reply:
Sounds reasonable. I'll do it as soon as time allows. Did you find anything else?

Reply:
This line is not backported:
https://github.com/cobbler/cobbler/blob/da8c2460f388603cc46e8002ece4d0fd0151c9c2/cobbler/items/item.py#L271